### DISCUSSION TITLE
Getting a runTime error while trying to reload cogs?

### DISCUSSION THREAD

**Discussion by enterCheesyNameHere** [Rapptz/discord.py#6019] [Q&A] [answered]
This is the code for the reload function I'm creating:  

```
     @commands.command(name="reload", help="Reloads a cog(s). Usage: $reload <cogName>")
     async def reload(self, ctx, cog):
        extensions = self.bot.extensions
        if cog == "all":
            for extension in extensions:
                self.bot.reload_extension(extension)
            await ctx.send("Done! All cogs have been reloaded.")
            return

        if cog in extensions:
            self.bot.reload_extension(cog)
            await ctx.send("Done! " + cog + " has been reloaded")
        else:
            await ctx.send("Unkown cog. Type '$listcogs' to see all cogs")
```

however when I type "$reload all" it gives me "RuntimeError: dictionary keys changed during iteration". When I type "$reload all" again it works, and this repeats over and over again, does anyone know why this is happening? I'm using VS code to run the code.

**Comment by ShineyDev** [comment, marked as answer]
The problem lines here are:

```python
            for extension in extensions:
                self.bot.reload_extension(extension)
```

[Bot.extensions](https://discordpy.readthedocs.io/en/latest/ext/commands/api.html#discord.ext.commands.Bot.extensions) is a [mapping](https://docs.python.org/3/library/types.html#types.MappingProxyType) and [mappings cannot be modified during iteration](https://stackoverflow.com/q/6777485).

You should modify the above lines to create a [copy](https://docs.python.org/3/library/types.html#types.MappingProxyType.copy) of the bot's extensions mapping before iteration.

**Comment by enterCheesyNameHere** [comment]
Its working perfectly, thanks!
